### PR TITLE
fix: Profile Details Overlap on User Dashboard After Registration

### LINKED ISSUE #2099: USER : Profile Details Overlap on User Dashboard After Registration
**Describe the bug**
Upon successfully registering as a user, the user dashboard displays an overlap issue in the Profile Details section, where the email address is superimposed on the profile image. This UI glitch reduces the user experience, and for long names, it results in poor UI, decreasing the readability.

**To Reproduce**

1. Setting up Talawa-api and Talawa-admin
2.Register and Login as a USER
3.Navigate to the USER dashboard
4.In cases of long names and email the UI glitch can be observed. 

**Expected behavior**
A column layout for the avatar image and other details, along with a tooltip to display the full text when it is overflowing, when hovered over.

**Actual behavior**
Overflowing of long names and emails out of the block giving rise to a poor UI.

**Screenshots**
Screenshot 2024-07-18 at 2 45 50 AM
Screenshot 2024-07-18 at 2 46 47 AM

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Saanvi26**
**What kind of change does this PR introduce?**

A tooltip to display the full email when it is overflowing, when hovered over.

**Issue Number:**

Fixes #2099 

**Did you add tests for your changes?**
Yes 

**Snapshots/Videos:**

Screenshot 2024-07-25 at 1 14 49 AM

Screenshot 2024-07-25 at 1 15 09 AM

Screenshot 2024-07-25 at 1 15 23 AM

**If relevant, did you update the documentation?**

NA

**Summary**

Issue #2099 
The user dashboard was displaying an overlap issue in the Profile Details section, where the email address was superimposed on the profile image. This UI glitch reduced the user experience ,decreasing the readability.
The motivation was to ensure smooth UI and add a tooltip for the users to view their email and name as they hover.

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
  - Introduced tooltip functionality for user details in the User Profile component, enhancing interactivity and providing contextual information on hover.

- **Bug Fixes**
  - Updated test cases to reflect changes in user details representation, improving accuracy in testing the User Profile component.

- **Style**
  - Adjusted layout styles for better alignment and responsiveness in the User Profile settings interface.

### CODE REVIEW

**Comment by Saanvi26** [comment]
Added package ' react-tooltip ' thats why test for ' Checks if no unauthorized files are changed ' failing

File: src/components/UserProfileSettings/UserProfile.tsx [hunk 1]
```diff
@@ -41,11 +42,26 @@ const UserProfile: React.FC<InterfaceUserProfile> = ({
...
-              <span style={{ fontWeight: '700', fontSize: '28px' }}>
-                {`${firstName}`.charAt(0).toUpperCase() +
-                  `${firstName}`.slice(1)}
+              <span
+                style={{ fontWeight: '700', fontSize: '28px' }}
+                data-tooltip-id="name"
+                data-tooltip-content={`${firstName} ${lastName}`}
+              >
+                {firstName.length > 10
+                  ? firstName.slice(0, 4) + '...'
```
Comment (varshith257) [inline src/components/UserProfileSettings/UserProfile.tsx:51]:
```suggestion
                  ? firstName.slice(0, 5) + '..'
```

File: src/components/UserProfileSettings/UserProfile.tsx [hunk 2]
```diff
@@ -41,11 +42,26 @@ const UserProfile: React.FC<InterfaceUserProfile> = ({
...
               </span>
-              <span data-testid="userEmail">{email}</span>
+              <ReactTooltip id="name" />
+              <span
+                data-testid="userEmail"
+                data-tooltip-id="email"
+                data-tooltip-content={email}
+              >
+                {email.length > 10
+                  ? email.slice(0, 3) + '...' + email.slice(email.indexOf('@'))
```
Comment (varshith257) [inline src/components/UserProfileSettings/UserProfile.tsx:61]:
```suggestion
                  ? email.slice(0, 4) + '..' + email.slice(email.indexOf('@'))
```

**Review by varshith257 — COMMENTED** [review]
Good to have this! two nits and good to merge

**Pull request merged**